src/OpenTelemetry.Shared/IOfflineStorage.cs

```diff
@@ -0,0 +1,45 @@
+// <copyright file="IOfflineStorage.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Shared
+{
+    /// <summary>
+    /// Persistent storage API.
+    /// </summary>
+    public interface IOfflineStorage
+    {
+        /// <summary>
+        /// Reads all blobs from the storage.
+        /// </summary>
+        /// <returns>All blobs from storage.</returns>
+        public IEnumerable<IOfflineBlob> GetBlobs();
+
+        /// <summary>
+        /// Reads the latest stored blob from the storage.
```

Comment:
the method name doesn't indicate that it is the latest one being returned.
Also latest means the one which was stored at the last?

Reply:
Yes. Latest one is the one which is stored at last. Its always good to send the most recent data first. Based on it added the summary. Should we rename this method?

Reply:
Its always good to send the most recent data first --> This is not a concern of this API. Its upto the user of this API to decide if recent one or oldest one is important for them.